frontend/lib/ui/artefact_dialog/artefact_dialog.dart

```diff
@@ -16,47 +17,59 @@ class ArtefactDialog extends ConsumerWidget {
 
   final int artefactId;
 
+  Column get _invalidArtefactErrorMessage {
+    return const Column(
+      children: [
+        DialogHeader(),
+        Expanded(
+          child: Center(
+            child: Text('Artefact not found. It may be that a'
+                ' newer version has been released already'),
+          ),
+        ),
+      ],
+    );
+  }
+
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final family = AppRoutes.familyFromContext(context);
-    final artefact =
-        ref.watch(familyArtefactsProvider(family)).requireValue[artefactId];
-
-    return SelectionArea(
-      child: Dialog(
-        child: SizedBox(
-          height: min(800, MediaQuery.of(context).size.height * 0.8),
-          width: min(1200, MediaQuery.of(context).size.width * 0.8),
-          child: Padding(
-            padding: const EdgeInsets.symmetric(
-              horizontal: Spacing.level5,
-              vertical: Spacing.level3,
-            ),
-            child: (artefact == null)
-                ? const Column(
-                    children: [
-                      DialogHeader(),
-                      Expanded(
-                        child: Center(
-                          child: Text('Artefact not found. It may be that a'
-                              ' newer version has been released already'),
-                        ),
+    final artefacts = ref.watch(familyArtefactsProvider(family));
+    return artefacts.when(
+      data: (artefacts) {
+        final artefact = artefacts[artefactId];
+        return SelectionArea(
+          child: Dialog(
+            child: SizedBox(
+              height: min(800, MediaQuery.of(context).size.height * 0.8),
+              width: min(1200, MediaQuery.of(context).size.width * 0.8),
+              child: Padding(
+                padding: const EdgeInsets.symmetric(
+                  horizontal: Spacing.level5,
+                  vertical: Spacing.level3,
+                ),
+                child: (artefact == null)
+                    ? _invalidArtefactErrorMessage
+                    : Column(
+                        crossAxisAlignment: CrossAxisAlignment.start,
+                        children: [
+                          ArtefactDialogHeader(artefact: artefact),
+                          const SizedBox(height: Spacing.level4),
+                          ArtefactDialogInfoSection(artefact: artefact),
+                          const SizedBox(height: Spacing.level4),
+                          Expanded(
+                            child: ArtefactDialogBody(artefact: artefact),
+                          ),
+                        ],
```

Comment:
This is the change to fix the async error.

Reply:
Thanks for fixing my error